contrib/test.sh

```diff
@@ -31,19 +31,29 @@ then
     exit 0
 fi
 
-# Test without any features first
-cargo test --verbose
+# Defaults / sanity checks
+cargo test
 
-# Test each feature
-for feature in ${FEATURES}
-do
-    cargo test --verbose --features="$feature"
-done
+if [ "$DO_FEATURE_MATRIX" = true ]
+then
+    # All features
+    cargo test --features="$FEATURES"
+
+    # Single features
+    for feature in ${FEATURES}
+    do
+        cargo test --features="$feature"
+    done
 
-# Also build and run each example to catch regressions
-cargo build --examples
-# run all examples
-run-parts ./target/debug/examples
+    # Run all the examples
+    cargo build --examples
+    cargo run --example htlc --features=compiler 
+    cargo run --example parse
+    cargo run --example sign_multisig
+    cargo run --example verify_tx > /dev/null
+    cargo run --example psbt
+    cargo run --example xpub_descriptors
```

Comment:
Why manually list examples instead of run-parts? Is it because of the extra args to each example? We need to remember to add examples here every time.

Reply:
Its so the script can be run locally, but yes as you say there is a maintenance cost. Is it worth it, I don't know? I configure `target-dir` to be a global build cache so there is no `target` directory when I build so the examples don't run.